src/backend/joanie/core/models/products.py

```diff
@@ -591,6 +587,43 @@ def main_invoice(self) -> dict | None:
         except ObjectDoesNotExist:
             return None
 
+    @property
+    def is_free(self):
+        """
+        Return True if the order is free.
+        """
+        return not self.total
+
+    @property
+    def has_payment_method(self):
+        """
+        Return True if the order has a payment method.
+        """
+        return (
+            self.credit_card is not None
+            and self.credit_card.initial_issuer_transaction_identifier is not None
+        )
+
+    @property
+    def has_contract(self):
+        """
+        Return True if the order has a contract.
+        """
+        try:
+            return self.contract is not None  # pylint: disable=no-member
+        except Contract.DoesNotExist:
+            return False
```

Comment:
I think this method is no useless as it's use has been replaced by `has_unsigned_contract`